subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/DependencyConstraintHandlerScope.kt

```diff
@@ -70,7 +96,7 @@ private constructor(
      * @return The dependency constraint.
      * @see [DependencyConstraintHandler.add]
      */
-    operator fun Configuration.invoke(dependencyConstraintNotation: Any): DependencyConstraint? =
+    operator fun Configuration.invoke(dependencyConstraintNotation: Any): DependencyConstraint =
```

Comment:
I didn't remember changing the nullability myself or by some tasks (maybe linting?), but the `DependencyConstraintHandlerDelegate` always returns non nullable `DependencyConstraint` due to Kotlin's type specification. This is also true for other methods in this class if you switch from a `constraints` call to the delegate call, see line 74/100. This line does not use `constraints.add` but `add` using the delegate (which is in fact `constraints`, but written in Kotlin).
Open question: Do you accept such source breaking (not binary breaking) change and do you want this change for the other methods too?

Reply:
We'll handle the nullability discrepancy separately.
Could you please revert this line and have the added function above return `DependencyConstraint?` too for uniformity?